src/InputWithLabel.jsx

```diff
@@ -0,0 +1,25 @@
+import { useRef, useEffect } from 'react';
+
+const InputWithLabel = ({ children, todoTitle, handleTitleChange }) => {
+  const inputRef = useRef();
+
+  useEffect(() => {
+    inputRef.current.focus();
+  });
+
+  return (
+    <>
+      <label htmlFor="todoTitle">{children}: </label>
+      <input
+        ref={inputRef}
+        type="text"
+        id="todoTitle"
+        name={children}
+        value={todoTitle}
+        onChange={handleTitleChange}
+        required />
+      <input type="submit" value="Add" /></>
+  );
+}
```

Comment:
- The reascon we use children as props in react is when you want the developers using your component to be able to insert any form of jsx into it https://react.dev/reference/react/Children I don't think using it here is ideal
- we want this inputWithLabel component to be reusable meaning that we can place it anywhere and use it so I would suggest the following below and my reasons why

```suggestion
const InputWithLabel = ({ label,
  type = "text",
  id,
  name,
  value,
  onChange,
  children }) => {
  const inputRef = useRef();

  useEffect(() => {
    inputRef.current.focus();
  });

  return (
    <>
      <label htmlFor={ id }>{ label }: </label>
      <input
        ref={ inputRef }
        type={ type }
        id={ id }
        name={ name }
        value={ value }
        onChange={ onChange }
        required />

      { children }
    </>
  );
};
```

with this you are making it reusable where people can pass props they normally pass to the input tag but including label, so anytime you want your `InputWithLabel` component, just call it 

You might have noticed the `{chidren}` props been isolated, that is like a hatch you have given other developers to put anything within your input, now they can use use your input to creat an autocomplete component, passing what they want via the children props

You also noticed I removed the submit input. it is not needed because the work of your input component is to show the user and input with a label , with this, we can have multiple inputs in a form and then the form submits the values from that input

Reply:
Hi! I understand why I need to remove submit. 
But I do not understand other changes, I believe they conflict with this lesson's instructions:
https://classes.codethedream.org/course/react-v2/grackle?week=6&lesson=Lesson+1.6%3A+Reusable+Components%2C+Imperative+React
```
Refactor Input with Label to use Component Composition
 Open /src/InputWithLabel.jsx
 Replace label prop with children so that any child node(s) are used as the label text
 Open /src/AddTodoForm.jsx
 Refactor the InputWithLabel component
 Remove the label prop
 Change the component to have an open/close tag instead of being self-closing
 Pass the text Title inside the component tags
 ```
 Could you please take a look at this lesson instructions, probably I got them wrong?